### DISCUSSION TITLE
Upgrade version of npm used by codespaces

### DISCUSSION THREAD

**Discussion by fvictorio** [community/community#33958] [Codespaces]
When I create a codespaces environment from a (node.js) repository, the installed version of npm is 6.14.16, which has EOLed for a while now.

**Comment by bjbennet** [comment]
Hi @fvictorio , you could add a `devcontainer.json` file to your project to update to the latest version of `npm` automatically upon creation of the codespace. The file would look like this:

{
"postCreateCommand": "npm update"
}

**Reply by fvictorio** [reply]
I use codespaces mainly for investigating reproduction repos with bug reports, so I usually open a new one each time, so this solution won't work for me.
